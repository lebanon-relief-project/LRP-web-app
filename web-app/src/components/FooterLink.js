import React from "react";
import styled from "styled-components";
import devices from "../styles/Devices";

const FooterLink = (props) => {
  const links = props.links.map((link, index) => (
    <StyledListLink key={index}>{link}</StyledListLink>
  ));
  return (
    <StyledFooterLink>
      <FooterLinkTitle>{props.title}</FooterLinkTitle>
      <StyledList>{links}</StyledList>
    </StyledFooterLink>
  );
};

const FooterLinkTitle = styled.h2`
  font-weight: bold;
`;

const StyledFooterLink = styled.div`
  padding: 1rem;
  margin: 2rem 0;
  @media (max-width: ${devices.mobile}) {
    width: 75%;
    padding: 0;
    margin: 0;
    order: 2;
  }
`;

const StyledList = styled.ul`
  list-style-type: none;
  padding: 0;
`;

const StyledListLink = styled.li`
  text-decoration-line: underline;
`;

export default FooterLink;
