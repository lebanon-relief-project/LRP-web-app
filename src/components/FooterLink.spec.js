import React from "react";
import { render } from "@testing-library/react";
import FooterLink from "./FooterLink";

describe("the Card component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderNavbar());

    expect(container).toMatchSnapshot();
  });
});

const renderNavbar = () => {
  return render(<FooterLink title="testTitle" links={["link1", "link2"]} />);
};
