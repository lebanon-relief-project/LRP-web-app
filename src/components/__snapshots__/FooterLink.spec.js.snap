// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Card component should match the snapshot 1`] = `
<div>
  <div
    class="sc-gtssRu PuCqs"
  >
    <h2
      class="sc-bdnylx gpJKAy"
    >
      testTitle
    </h2>
    <ul
      class="sc-dlnjPT cQMtRp"
    >
      <li
        class="sc-hKFyIo igLYdx"
      >
        link1
      </li>
      <li
        class="sc-hKFyIo igLYdx"
      >
        link2
      </li>
    </ul>
  </div>
</div>
`;
